src/main/java/randoop/condition/SpecificationCollection.java

```diff
@@ -0,0 +1,312 @@
+package randoop.condition;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import randoop.compile.SequenceClassLoader;
+import randoop.compile.SequenceCompiler;
+import randoop.condition.specification.OperationSignature;
+import randoop.condition.specification.OperationSpecification;
+import randoop.reflection.TypeNames;
+import randoop.util.Log;
+import randoop.util.MultiMap;
+
+/**
+ * A collection of {@link OperationSpecification} objects. Also a map from an {@link
+ * AccessibleObject} reflection object to the {@link OperationConditions} for the corresponding
+ * {@link randoop.operation.TypedClassOperation}.
+ *
+ * <p>The {@link SpecificationCollection} should be constructed from the specification input before
+ * the {@link randoop.reflection.OperationModel} is created.
+ *
+ * <p>This class stores the {@link OperationSpecification} objects, and only constructs the
+ * corresponding {@link OperationConditions} on demand. This lazy-strategy avoids building
+ * conditions methods for specifications that are not used.
+ */
+public class SpecificationCollection {
+
+  /** The map from reflection objects to the corresponding {@link OperationSpecification} */
+  private final Map<AccessibleObject, OperationSpecification> specificationMap;
+
+  /** The map from method signatures to methods with that signature and specifications */
```

Comment:
What does "and specifications" mean?  Do you mean that the method must have a specification?  Clarify the wording in any event.